core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -164,7 +164,7 @@ private FileIO io(String location) {
     return io;
   }
 
-  private static String implFromLocation(String location) {
+  public static String implFromLocation(String location) {
```

Comment:
@rdblue do you have concern about making this public?

Reply:
The name of the method sounds like something we would want to keep private.

Reply:
yes that's why I am a bit hesitated, but I think it also makes sense that `ResolvingFileIO` allows the caller to know what is the `FileIO` it will resolve to for each location.

Reply:
I agree. What about adding a new method called `ioClass` like this?

```
public Class<? extends FileIO> ioClass(String location) {
  return io(location).getClass();
}
```

We could also rename `implFromLocation` to be `ioClassName` and open it but it won't be possible to check if a particular file IO is a subtype of `HadoopFileIO`. I believe the existing code can handle that.

Reply:
If we have `Class<? extends FileIO>`, we can use `isAssignableFrom` to check types.